test/unit/controllers/PeerSelectControllerSpec.js

```diff
@@ -29,229 +27,227 @@ describe('PeerSelectController', function () {
+    this.promise = {
+      promiseFlag: false,
       then: function (f) {
-        $promiseFlag = true
+        this.$promiseFlag = true
         f()
       }
     }
 
-    $EService = {
-      confirm: function (data) {
-        $promiseData = data
-        return $promise
+    var promise = this.promise
+
+    this.$q = {
+      when: function () {
+        return promise
       }
     }
 
-    $q = {
-      when: function () {
-        return $promise
+    this.ErrorService = {
+      $promiseData: {},
+      confirm: function (data) {
+        this.$promiseData = data
+        return promise
       }
     }
 
+    var $q = this.$q
+    var AppPeersManager = this.AppPeersManager
+    var $modalInstance = this.$modalInstance
+    var ErrorService = this.ErrorService
+
     inject(function (_$controller_, _$rootScope_) {
-      $controller = _$controller_
-      $scope = _$rootScope_.$new()
+      this.$controller = _$controller_
+      this.$scope = _$rootScope_.$new()
+      var $scope = this.$scope
+      var $controller = this.$controller
+
+      // The controller is created in the test in order to test different initial content of scope variables.
+      this.createController = function () {
+        $controller('PeerSelectController', {
+          $scope: $scope,
+          $modalInstance: $modalInstance,
+          $q: $q,
+          AppPeersManager: AppPeersManager,
+          ErrorService: ErrorService
+        })
+      }
     })
   })
 
   it('initialises properties', function (done) {
-    createController()
+    this.createController()
 
     // Set timer to give the controller time to resolve.
     setTimeout(function () {
-      expect($scope.selectedPeers).toBeDefined()
-      expect($scope.selectedPeersIDs).toBeDefined()
-      expect($scope.selectedCount).toBeDefined()
-    }, timeoutTime)
+      expect(this.$scope.selectedPeers).toBeDefined()
+      expect(this.$scope.selectedPeersIDs).toBeDefined()
+      expect(this.$scope.selectedCount).toBeDefined()
+    }, this.waitOneSecond)
 
     done()
   })
 
   it('compiles with a shareLinkPromise that resolves', function (done) {
     var expected = 'testURL'
-    $scope.shareLinkPromise = {
+    var waitOneSecond = this.waitOneSecond
+    this.$scope.shareLinkPromise = {
       then: function (resolve, reject) {
-        setTimeout(resolve(expected), timeoutTime)
+        setTimeout(resolve(expected), waitOneSecond)
       }
     }
-    createController()
+    this.createController()
 
     setTimeout(function () {
-      expect($scope.shareLink.loading).toBe(true)
-      expect($scope.shareLink.url).not.toBeDefined()
+      expect(this.$scope.shareLink.loading).toBe(true)
+      expect(this.$scope.shareLink.url).not.toBeDefined()
       setTimeout(function () {
-        expect($scope.shareLink.url).toBe(expected)
-      }, timeoutTime)
-    }, timeoutTime)
+        expect(this.$scope.shareLink.url).toBe(expected)
+      }, waitOneSecond)
+    }, waitOneSecond)
```

Comment:
Move the inner setTimeout to a helper function at least. So it reads something like `setTimeout(innerTimeout, oneSecond)` with a better named `innerTimeout = function() {expect(/*...*/)}`.